benchmark-overhead/src/test/java/io/opentelemetry/results/CsvPersister.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.results;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Consumer;
+
+class CsvPersister implements ResultsPersister {
+
+  private final Path resultsFile;
+
+  public CsvPersister(Path resultsFile) {this.resultsFile = resultsFile;}
+
+  @Override
+  public void write(List<AppPerfResults> results) {
+
+    ensureFileCreated(results);
+
+    StringBuffer sb = new StringBuffer().append(System.currentTimeMillis() / 1000);
+    doSorted(results, result -> {
+      sb.append(",").append(result.startupDurationMs);
+      sb.append(",").append(result.heapUsed.min);
+      sb.append(",").append(result.heapUsed.max);
+      sb.append(",").append(result.getTotalAllocatedMB());
+      sb.append(",").append(result.totalGCTime);
+      sb.append(",").append(result.maxThreadContextSwitchRate);
+      sb.append(",").append(result.iterationAvg);
+      sb.append(",").append(result.iterationP95);
+      sb.append(",").append(result.requestAvg);
+      sb.append(",").append(result.requestP95);
+      sb.append(",").append(result.peakThreadCount);
+    });
+    sb.append("\n");
+    try {
+      Files.writeString(resultsFile, sb.toString(), StandardOpenOption.APPEND);
+    } catch (IOException e) {
+      throw new RuntimeException("Error writing csv content", e);
+    }
+  }
+
+  private void ensureFileCreated(List<AppPerfResults> results) {
+    if (Files.exists(resultsFile)) {
+      return;
+    }
+    StringBuffer sb = new StringBuffer("timestamp");
+    doSorted(results, result -> {
+      String agent = result.getAgentName();
+      sb.append(",").append(agent).append(":startupTimeMs");
+      sb.append(",").append(agent).append(":minHeapUsed");
+      sb.append(",").append(agent).append(":maxHeapUsed");
+      sb.append(",").append(agent).append(":totalAllocatedMB");
+      sb.append(",").append(agent).append(":totalGCTime");
+      sb.append(",").append(agent).append(":maxThreadContextSwitchRate");
+      sb.append(",").append(agent).append(":iterationAvg");
+      sb.append(",").append(agent).append(":iterationP95");
+      sb.append(",").append(agent).append(":requestAvg");
+      sb.append(",").append(agent).append(":requestP95");
+      sb.append(",").append(agent).append(":peakThreadCount");
+    });
+    sb.append("\n");
+    try {
+      Files.writeString(resultsFile, sb.toString());
+    } catch (IOException e) {
+      throw new RuntimeException("Error creating csv output stub", e);
+    }
```

Comment:
Please extract this to something a la `writeHeader` or something.

And it now seems that it is indeed expected to write one long line. Please add some comment about that. I bet everybody will be confused that a list of items is all written as one single line.

Reply:
Factoring out the named method makes sense.
The long line shouldn't really be surprising for a csv file tho should it?

Reply:
when I see a loop inside "write to CSV" method, my first expectation is that each loop iteration writes one line. I bet I am not the only one.

Reply:
🤷🏻 Ok I'll add a comment.